pwiz_tools/Skyline/Model/Lib/BlibData/BlibDb.cs

```diff
@@ -244,6 +245,157 @@ public void Write()
             return CreateLibraryFromSpectra(librarySpec, listSpectra, libraryName, progressMonitor, ref status);
         }
 
+
+        private class SpectrumInserter : IDisposable
+        {
+
+            // ReSharper disable LocalizableElement
+            static string _insertSpectraSql = "INSERT INTO RefSpectra (id, peptideSeq, peptideModSeq, precursorCharge, " +
+                                              "precursorAdduct, precursorMZ, moleculeName, chemicalFormula, inchiKey, " +
+                                              "otherKeys, prevAA, nextAA, copies, numPeaks, ionMobility, collisionalCrossSectionSqA, " +
+                                              "ionMobilityHighEnergyOffset, ionMobilityType, retentionTime, fileId, SpecIDinFile, score, scoreType) " +
+                                              "VALUES (" + string.Join(",", Enumerable.Repeat("?", 23)) + ")";

```

Comment:
You might want to add "; select last_insert_rowid();" to the end of this SQL statement.
The way that you are doing it now where you first query for the highest existing rowid and keep incrementing that number every time you insert a row is a fine way to do it too.